mongo/mongo.go

```diff
@@ -0,0 +1,85 @@
+package mongo
+
+import (
+	"context"
+
+	"github.com/ONSdigital/dp-healthcheck/healthcheck"
+	"github.com/ONSdigital/dp-legacy-cache-api/config"
+	"github.com/ONSdigital/dp-legacy-cache-api/models"
+	mongoHealth "github.com/ONSdigital/dp-mongodb/v3/health"
+	mongoDriver "github.com/ONSdigital/dp-mongodb/v3/mongodb"
+	"github.com/ONSdigital/log.go/v2/log"
+	"go.mongodb.org/mongo-driver/bson"
+)
+
+type Mongo struct {
+	mongoDriver.MongoDriverConfig
+
+	Connection   *mongoDriver.MongoConnection
+	healthClient *mongoHealth.CheckMongoClient
+}
+
+// NewMongoStore creates a connection to mongo database
+func NewMongoStore(_ context.Context, cfg config.MongoConfig) (m *Mongo, err error) {
+	m = &Mongo{MongoDriverConfig: cfg}
+
+	m.Connection, err = mongoDriver.Open(&m.MongoDriverConfig)
+
+	if err != nil {
+		return nil, err
+	}
+	databaseCollectionBuilder := map[mongoHealth.Database][]mongoHealth.Collection{
+		mongoHealth.Database(m.Database): {
+			mongoHealth.Collection(m.ActualCollectionName(config.DatasetsCollection)),
+		},
+	}
+
+	m.healthClient = mongoHealth.NewClientWithCollections(m.Connection, databaseCollectionBuilder)
+
+	return m, nil
+}
+
+// GetDataSets reads all records in the connected database collection
```

Comment:
That's not entirely true - on line 48 you're specifying the collection, i.e. you're collecting from the 'Datasets' collection.